nautobot/extras/querysets.py

```diff
@@ -163,34 +165,71 @@ def _get_config_context_filters(self):
 class DynamicGroupQuerySet(RestrictedQuerySet):
     """Queryset for `DynamicGroup` objects that provides a `get_for_object` method."""
 
-    # FIXME(jathan): Ideally replace this iteration with a reversible Q object
-    # of some sort.
-    def get_for_object(self, obj):
-        """Return all `DynamicGroup` assigned to the given object."""
+    def get_list_for_object(self, obj, skip_cache=False):
+        """
+        Return a list of `DynamicGroup` assigned to the given object. As opposed to `get_for_object`
+        which will return a queryset but that is an additional query to the DB when you may just
+        want a list.
+
+        Args:
+            obj: The object to seek dynamic groups membership by.
+            skip_cache: If True, skip the cache and query the database directly.
+        """
         if not isinstance(obj, Model):
             raise TypeError(f"{obj} is not an instance of Django Model class")
 
         # Get dynamic groups for this content_type using the discrete content_type fields to
         # optimize the query.
-        # FIXME(jathan): Try to get the number of queries down. This will scale poorly.
-        # TODO(jathan): 1 query
-        eligible_groups = self.filter(
-            content_type__app_label=obj._meta.app_label, content_type__model=obj._meta.model_name
-        ).select_related("content_type")
+        eligible_groups = self._get_eligible_dynamic_groups(obj, skip_cache=skip_cache)
 
         # Filter down to matching groups.
         my_groups = []
-        # TODO(jathan): 3 queries per DynamicGroup instance
-        for dynamic_group in eligible_groups.iterator():
-            if dynamic_group.members.filter(pk=obj.pk).values_list("pk", flat=True).exists():
-                my_groups.append(dynamic_group.pk)
+        for dynamic_group in list(eligible_groups):
+            if dynamic_group.am_i_a_member(obj, skip_cache=skip_cache):
+                my_groups.append(dynamic_group)
+
+        return my_groups
+
+    def get_for_object(self, obj, skip_cache=False):
```

Comment:
Any concern about breaking changes in that `get_for_object` used to always perform the query, whereas now it'll use the cache unless you explicitly specify `skip_cache=True`? (In other words, should we invert the default for `skip_cache` to stay backwards-compatible?)

Reply:
This added some drop-in improvements to some of the places we use it (GraphQL resolver, etc). But I'm not opposed. @nautobot/core?